### PR TITLE
Run example tests concurrently with existing tests in PRs

### LINKED ISSUE #6627: Make Examples tests pass
Examples tests are currently failing, see e.g. https://github.com/openscad/openscad/actions/runs/21781289454

Get this sorted so the tests are green.

### PR DESCRIPTION
**Pull request by coryrc**
Fixes #6627

### CODE REVIEW

**Comment by coryrc** [comment]
Oops, I didn't rerun *all* the example tests after making changes. Fixing.

Also, the "branch protection" will need to be updated with the new names for the test workflows, or I could create a dummy job with the same name as the old one to run successfully only when the tests pass.

File: .github/workflows/linux-tests.yml [hunk 1]
```diff
@@ -18,39 +17,79 @@ jobs:
...
+          retention-days: 1
 
+  test:
+    needs: build
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-22.04, ubuntu-24.04 ]
+        qt: [ qt5, qt6 ]
+        task: [ "test", "test_examples test coverage" ] # examples are faster than non-examples, so they get coverage
```
Comment (kintel) [inline .github/workflows/linux-tests.yml]: The coverage stuff has probably bitrotted a bit, but the intent is to run coverage over the whole test run to look for untested code. As such, it should probably run on the main test run, as that's where we cover corner cases.

File: examples/Advanced/surface_image.scad [hunk 2]
```diff
@@ -6,8 +6,8 @@ echo(version=version());
 // those as 3 stacked layers.
 
 for (a = [1, 2, 3])
-    color([a/6 + 0.5, 0, 0])
-       linear_extrude(height = 2 * a, convexity = 10)
+    color([a/6 + 0.5, 0, 0]) translate([0,0,2*(a-1)])
+       linear_extrude(height = 2, convexity = 10)
```
Comment (kintel) [inline examples/Advanced/surface_image.scad]: I would make height 2.1, and subtract 0.1 in translate, to disambiguate between 3 disconnected slabs vs. one solid object

**Comment by kintel** [comment]
> 
> Also, the "branch protection" will need to be updated with the new names for the test workflows, or I could create a dummy job with the same name as the old one to run successfully only when the tests pass.

Dummy job makes sense, and we can split it up later if needed.

File: .github/workflows/linux-tests.yml (hunk 1 above)
Comment (coryrc) [reply .github/workflows/linux-tests.yml]: Coverage doesn't count the regression tests AFAICT -- how could it? It just does:

```
		echo "Generating code coverage report..."
		uv run --project "$REPO_ROOT/.github/ci/openscad-coverage" gcovr -r ../src CMakeFiles/OpenSCADLibInternal.dir $PARALLEL_GCOVR --html --html-details --sort uncovered-percent -o coverage.html
```

However, I now see the workspace upload and time to reinstall apt packages takes 3 minutes to run; and so do the example tests. So there's no reason to spin up separate VM.

Comment (kintel) [reply .github/workflows/linux-tests.yml]: > Coverage doesn't count the regression tests AFAICT -- how could it? It just does:
> 
We run CMake with `-DPROFILE=ON`, which instruments the code to collect coverage info and output into a database, which is later read by `gcovr`.

File: examples/Advanced/surface_image.scad (hunk 2 above)
Comment (coryrc) [reply examples/Advanced/surface_image.scad]: Is that because this serves as an example and you think people should write their code this way?

File: .github/workflows/linux-tests.yml (hunk 1 above)
Comment (coryrc) [reply .github/workflows/linux-tests.yml]: I added a comment to the script with that info, so I won't ask again :)

File: examples/Advanced/surface_image.scad (hunk 2 above)
Comment (kintel) [reply examples/Advanced/surface_image.scad]: It's because coincident surfaces yield an undefined result, which may change between Manifold versions. In practice we always have to use some overlap to disambiguate, so it's also best practice when using OpenSCAD.

Comment (coryrc) [reply examples/Advanced/surface_image.scad]: Along with the required modifications, I added a comment:

`   overlap = a==1 ? 0 : 0.1; // Overlapping ensures one mesh.`

**Comment by coryrc** [comment]
And currently we just no longer run throwntogether for example009, which solves the problem. Preview and render do get run.

**Review by kintel — APPROVED** [review]
